### ISSUE TITLE
False positive from no-descending-specificity

### ISSUE THREAD

**Issue by paulschreiber** [stylelint/stylelint#5002]
There are false positives for `no-descending-specificity` in the following case:

### Which rule, if any, is the bug related to?

`no-descending-specificity`

### What code is needed to reproduce the bug?
```css
.contact-form,
.gform_wrapper .gform_body {
	select,
	input[type='text'] {
		font-size: 16px;
	}

	select {
		appearance: none;
	}
}
```

### Regression 
* If you use either `.contact-form {` or `.gform_wrapper .gform_body {`, this works as expected.

* If you use 
```css
.gform_wrapper .gform_body,
.contact-form {
```

you get four errors instead of one.

* Reversing the order of the two select blocks has no effect.

> What stylelint configuration is needed to reproduce the bug?
```json
{
	"extends": "stylelint-config-standard",
	"rules": {
		"indentation": "tab",
		"at-rule-no-unknown": null
	}
}
```

### Which version of stylelint are you using?

13.7.2

### How are you running stylelint: CLI, PostCSS plugin, Node.js API?

grunt

### Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Possibly.

### What did you expect to happen?

No warnings to be flagged.

### What actually happened (e.g. what warnings or errors did you get)?

e.g. "The following warnings were flagged:"

```shell
Running "stylelint:src" (stylelint) task

assets/css/src/_form.scss
 8:2  ✖  Expected selector ".contact-form select" to come before selector          no-descending-specificity
         ".gform_wrapper .gform_body select"

✖ 1 problem
```

**Comment by jeddy3** [comment]
Thanks for the report and for using the template.

```scss
.contact-form,
.gform_wrapper .gform_body {
	select,
	input[type='text'] {
		font-size: 16px;
	}

	select {
		appearance: none;
	}
}
```

Will unwrapped to:

```css
.contact-form select,
.contact-form input[type=text],
.gform_wrapper .gform_body select,
.gform_wrapper .gform_body input[type=text] {
  font-size: 16px;
}
.contact-form select,
.gform_wrapper .gform_body select {
  appearance: none;
}
```

Where `.gform_wrapper .gform_body select` has a higher specificity than `.contact-form select`.

If you're nesting your code in this way, [the docs recommend](https://stylelint.io/user-guide/rules/no-descending-specificity#dom-limitations) either restructuring your code or turning off the rule for this code snippet.
